compiler/src/lexer.mll

```diff
@@ -134,10 +134,10 @@ rule main = parse
   | '"' (([^'"' '\\']|'\\' _)* as s) '"' { STRING (unescape (L.of_lexbuf lexbuf) s) }
 
   (* Why this is needed *)
-  | ((*'-'?*) digit+) as s   
-      {INT s} 
+  | ((*'-'?*) digit+('_'digit+)*) as s   
+      {INT Str.(global_replace (regexp "_") "" s)} 
```

Comment:
You need to keep the exact representation.

Reply:
Changed in last commit, the function that parse `int_representation` type delete the '_' before converting to `Z.t`